src/network/networkd-route.c

```diff
@@ -77,15 +77,20 @@ int route_new(Route **ret) {
         return 0;
 }
 
-int route_new_static(Network *network, unsigned section, Route **ret) {
+int route_new_static(Network *network, const char *filename, unsigned section_line, Route **ret) {
+        _cleanup_network_config_section_free_ NetworkConfigSection *n = NULL;
         _cleanup_route_free_ Route *route = NULL;
         int r;
 
         assert(network);
         assert(ret);
 
-        if (section) {
-                route = hashmap_get(network->routes_by_section, UINT_TO_PTR(section));
+        if (section_line) {
```

Comment:
Again, a minor thing, but worth changing: we don't use integers directly in `if` — we do that only for `bool`s and pointers. So in this case this should have been `if (section_line > 0)`. But I'd prefer this changes, as I requested before, to `if (filename)`. That seems more natural, because filename is the more important of the (filename, line number) pair.